db/collection.go

```diff
@@ -451,20 +458,17 @@ func (t *Txn) Discard() {
 	t.discarded = true
 }
 
-func getInstanceID(t []byte) core.InstanceID {
+func getInstanceID(t []byte) (id core.InstanceID, err error) {
 	partial := &struct {
 		ID *string `json:"_id"`
 	}{}
 	if err := json.Unmarshal(t, partial); err != nil {
-		log.Fatalf("error when unmarshaling json instance: %v", err)
+		return id, fmt.Errorf("error unmarshaling json instance: %v", err)
 	}
 	if partial.ID == nil {
-		log.Fatal("invalid instance: doesn't have an _id attribute")
-	}
-	if *partial.ID != "" && !core.IsValidInstanceID(*partial.ID) {
```

Comment:
This looked to be an impossible condition: _an empty string and an invalid instance_, since the only way an instance is invalid is if it's not an empty string! I think the validity check used to check for UUID type or something.